app/org/sagebionetworks/bridge/models/healthdata/HealthDataAttachment.java

```diff
@@ -0,0 +1,30 @@
+package org.sagebionetworks.bridge.models.healthdata;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+
+import org.sagebionetworks.bridge.dynamodb.DynamoHealthDataAttachment;
+import org.sagebionetworks.bridge.json.BridgeTypeName;
+
+/**
+ * Metadata record about health data attachments. This is used to keep track of attachments in external storage
+ * (generally S3) and the actual health data record.
+ */
+@BridgeTypeName("HealthDataAttachment")
+@JsonDeserialize(as = DynamoHealthDataAttachment.class)
+public interface HealthDataAttachment {
+    /**
+     * Attachment ID. This is used as both the unique attachment ID and the S3 key in the attachments bucket to
+     * guarantee filename uniqueness in S3.
+     */
+    String getId();
+
+    /** Record ID of the health data record this file is attached to. */
+    String getRecordId();
```

Comment:
Are we expecting one record many attachments?  What can be a valid case that we want to store multiple files for one health record.  I am wondering if we could simplify it to a 1:1 relationship.  Then we don't need the index on record id.  I am usually not a big fan of mixing up writes with reads especially for tables receiving batch updates.